src/main/java/com/ibm/devops/notification/BuildListener.java

```diff
@@ -49,17 +50,80 @@ private void handleEvent(AbstractBuild r, TaskListener listener, String phase) {
         EnvVars envVars = EventHandler.getEnv(r, listener, printStream);
         String webhook = EventHandler.getWebhookFromEnv(envVars);
         Result result = r.getResult();
-        Boolean isRelevant = EventHandler.isRelevant(notifier, phase, result);
 
-        if(isRelevant) {
+        // OTC Notifier
+        if(EventHandler.isRelevant(notifier, phase, result)) {
             String resultString = null;
-
             if(result != null){
                 resultString = result.toString();
             }
 
             JSONObject message = MessageHandler.buildMessage(r, envVars, phase, resultString);
-            MessageHandler.postToWebhook(webhook, message, printStream);
+            MessageHandler.postToWebhook(webhook, false, message, printStream);
+        }
+        
+        // deployable mapping
+        if(EventHandler.shouldPostDeployableMappingMessage(notifier, phase, result)) {
+            String resultString = null;
+            if(result != null){
+                resultString = result.toString();
+            }
+            
+            if (validateEnvVariables(envVars, printStream)) {
+            	printStream.println("[IBM Cloud DevOps] Building Deployable Message.");
+                JSONObject message = MessageHandler.buildDeployableMappingMessage(envVars, printStream);
+                
+                MessageHandler.postToWebhook(webhook, true, message, printStream);	
+            } else {
+            	printStream.println("[IBM Cloud DevOps] Not sending Deployable Message due to missing required property.");
+            }
         }
     }
+    
+    public static boolean validateEnvVariables(EnvVars envVars, PrintStream printStream) {
+    	Boolean valid = true;
+    	if(envVars != null) {
+    		// required params
+    		String cf_api = envVars.get("CF_API");
+    		String cf_org = envVars.get("CF_ORG");
+    		String cf_space = envVars.get("CF_SPACE");
+    		String cf_app = envVars.get("CF_APP");
+    		String cf_user = envVars.get("CF_CREDS_USR");
+    		String cf_pwd = envVars.get("CF_CREDS_PSW");
+    		String webhook = envVars.get("IBM_CLOUD_DEVOPS_WEBHOOK_URL");
+    		//backward compatibility
```

Comment:
I guess config steps are linked from the readme
https://console.ng.bluemix.net/docs/services/ContinuousDelivery/toolchains_integrations.html#jenkins
and will be modified accordingly
https://github.ibm.com/org-ids/ua-issues/issues/463

the main target here was to get rid asap of the old cf icd plugin.
getting rid of the variables might be a separate issue / enhancement

thoughts?